### PR TITLE
Raise when doctest's only function is undefined

### PR DESCRIPTION
**Pull request by RudolfMan**
as proposed in https://groups.google.com/g/elixir-lang-core/c/zlM7pvqCXzE

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/doc_test.ex [hunk 1]
```diff
@@ -234,30 +234,50 @@ defmodule ExUnit.DocTest do
...
+    case only -- fun_arities do
+      [] ->
+        filtered_tests
+
+      [_ | _] = undefined_fun_arities ->
+        functions = Enum.map_join(undefined_fun_arities, ", ", fn {f, a} -> "#{f}/#{a}" end)
+
+        raise Error,
+          module: module,
+          message: "undefined or private function(s): #{functions} in module #{inspect(module)}"
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/doc_test.ex:277]:
```suggestion
        functions = Enum.map_join(undefined_fun_arities, ", ", fn {f, a} -> Exception.format_mfa(module, f, a) end)

        raise Error,
          module: module,
          message: "undefined or private function(s) given to doctest: #{functions}"
```

**Review by josevalim — COMMENTED** [review]
Thank you! Just one last tiny comment. You may need to run the formatter and update tests.

File: lib/ex_unit/lib/ex_unit/doc_test.ex (hunk 1 above)
Comment (RudolfMan) [reply lib/ex_unit/lib/ex_unit/doc_test.ex:277]: I think this makes it less readable when there are multiple functions undefined. They all belong to the same module. 

May be, at least, list them each on new line padded with 4 spaces, like:
```
** (ExUnit.DocTest.Error) <file_name>: undefined or private function(s) given to doctest:

    Module.foo/0
    Module.bar/1
    
<stacktrace>
```

Comment (josevalim) [reply lib/ex_unit/lib/ex_unit/doc_test.ex:277]: Multi line looks great. Your call!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**